ppod.py

```diff
@@ -19,4 +22,46 @@ def lambda_handler(event: dict, context: object) -> str:
         logger.info(
             "Sentry DSN found, exceptions will be sent to Sentry with env=%s", env
         )
-    return "lambda"
+    file_count = 0
+    try:
+        bucket = event["queryStringParameters"]["bucket"]
+        s3_files = filter_files_in_bucket(
+            bucket,
+            event["queryStringParameters"]["file_type"],
+            event["queryStringParameters"]["key_prefix"],
+        )
+        for s3_file in s3_files:
+            logger.info(f"Processing file: {s3_file}")
+            s3_file_content = smart_open.open(f"s3://{bucket}/{s3_file}", "rb")
+            files = extract_files_from_tar(s3_file_content)
+            for file in files:
+                file  # do a thing
+                file_count += 1
+    except KeyError:
+        pass
+    return f"Files processed: {file_count}"
+
+
+def extract_files_from_tar(tar_file: IO[bytes]) -> Generator[IO[bytes], None, None]:
+    tar = tarfile.open(fileobj=tar_file)
+    for member in tar.getmembers():
+        file = tar.extractfile(member)
+        if file is not None:
```

Comment:
Another place where I don't think we want to let this fail silently. If `extractfile()` returns `None` for any member of the tarfile, something has likely gone wrong in the export process and we need to raise an exception for further investigation.